sdks/java/harness/src/test/java/org/apache/beam/fn/harness/state/MultimapUserStateTest.java

```diff
@@ -179,6 +184,66 @@ public void testKeys() throws Exception {
     assertThrows(IllegalStateException.class, () -> userState.keys());
   }
 
+  @Test
+  public void testEntries() throws Exception {
+    FakeBeamFnStateClient fakeClient =
+        new FakeBeamFnStateClient(
+            ImmutableMap.of(
+                createMultimapEntriesStateKey(),
+                KV.of(
+                    KvCoder.of(ByteArrayCoder.of(), IterableCoder.of(StringUtf8Coder.of())),
+                    asList(KV.of(A1, asList("V1", "V2")), KV.of(A2, asList("V3"))))));
+    MultimapUserState<byte[], String> userState =
+        new MultimapUserState<>(
+            Caches.noop(),
+            fakeClient,
+            "instructionId",
+            createMultimapKeyStateKey(),
+            ByteArrayCoder.of(),
+            StringUtf8Coder.of());
+
+    assertArrayEquals(A1, userState.entries().iterator().next().getKey());
+    assertThat(
+        StreamSupport.stream(userState.entries().spliterator(), false)
+            .map(entry -> KV.of(ByteString.copyFrom(entry.getKey()), entry.getValue()))
+            .collect(Collectors.toList()),
+        containsInAnyOrder(
+            KV.of(ByteString.copyFrom(A1), "V1"),
+            KV.of(ByteString.copyFrom(A1), "V2"),
+            KV.of(ByteString.copyFrom(A2), "V3")));
+
+    userState.put(A1, "V4");
+    assertThat(
```

Comment:
should we keep a handle to previous userSTate.entries() result or iterator we partially iterated and verify that it is still valid and doesn't contain the new adds/removes?

Reply:
Good idea.